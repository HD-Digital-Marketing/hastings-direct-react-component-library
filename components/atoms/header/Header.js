import React from 'react';
import './header.scss';
import classNames from 'classnames';

const Header = props => {

    const {
        children,
        headerType = 'h1',
        className = '',
        theme
    } = props;
    
    let header = '';

    let headerClass = classNames('a-header', className, {
        'a-header--primary': theme === 'primary',
        'a-header--primary-xl': theme === 'primary-xl',
        'a-header--primary-on-dark': theme === 'primary-dark',
        'a-header--primary-on-dark-xl': theme === 'primary-dark-xl',
        'a-header--secondary': theme === 'secondary',
        'a-header--secondary-xl': theme === 'secondary-xl',
        'a-header--secondary-on-dark': theme === 'secondary-dark',
        'a-header--secondary-on-dark-xl': theme === 'secondary-dark-xl',

    });

    switch (headerType) {
        case 'h1':
            header = <h1 className={headerClass}>{children}</h1>
            break;
        case 'h2':
            header = <h2 className={headerClass}>{children}</h2>
            break;
        case 'h3':
            header = <h3 className={headerClass}>{children}</h3>
            break;
        case 'h4':
            header = <h4 className={headerClass}>{children}</h4>
            break;
        case 'h5':
            header = <h5 className={headerClass}>{children}</h5>
            break;
        case 'h6':
            header = <h6 className={headerClass}>{children}</h6>
            break;
        default:
            header = <h1 className={headerClass}>{children}</h1>
    }

    return (
        header
    )

}

export default Header;