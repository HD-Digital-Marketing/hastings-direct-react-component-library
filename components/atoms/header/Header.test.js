import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('Check the title renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Header headerType="h3" theme="primary">I am a h3 title</Header>)
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})