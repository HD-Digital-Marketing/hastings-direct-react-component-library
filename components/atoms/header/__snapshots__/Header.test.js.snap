// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the title renders it should match the snapshot 1`] = `
<h3
  className="a-header a-header--primary"
>
  I am a h3 title
</h3>
`;
